packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -39,41 +43,64 @@ export class AtomicSearchInterface {
   @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'silent';
+  @Prop() logLevel?: LogLevel;
   @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({reflect: true}) language = 'en'; // TODO: make watchable and update i18next language on change
   @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
+  @Event() ready!: EventEmitter;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
-  private initialized = false;
-  private afterInitializationCallbacks: (() => void)[] = [];
+  private systemsToInitialize = {
+    i18next: false,
+    headless: false,
+  };
 
   componentWillLoad() {
+    this.i18n
+      .use(Backend)
+      .init({
+        debug: this.logLevel === 'debug',
+        lng: this.language,
+        fallbackLng: ['en'],
+        backend: {
+          loadPath: `${getAssetPath('./lang/')}{{lng}}.json`,
+        } as BackendOptions,
+      })
+      .then(() => this.systemInitialized('i18next'));
+
     if (this.sample) {
-      this.initialize(HeadlessEngine.getSampleConfiguration());
+      // Mimics calling the async "initialize" method from the DOM
```

Comment:
Putting the call in `componentDidLoad` is an alternative, but it makes trigger a warning that it makes the component re-render (which is normal since it renders once for the loading state, and once when the engine/i18n are defined)